perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_member_attribute_def_def_reasonToSuspend.java

```diff
@@ -0,0 +1,92 @@
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeValueException;
+import cz.metacentrum.perun.core.api.exceptions.WrongReferenceAttributeValueException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.MemberAttributesModuleAbstract;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Message with a reason to suspend a member.
+ *
+ * @author Metodej Klang <metodej.klang@gmail.com>
+ */
+public class urn_perun_member_attribute_def_def_reasonToSuspend extends MemberAttributesModuleAbstract {
+
+	private final static Logger log = LoggerFactory.getLogger(urn_perun_member_attribute_def_def_reasonToSuspend.class);
+
+	private final Pattern statusChange = Pattern.compile("Member:\\[id=<([0-9]+)>.*] ([a-z]+)", Pattern.DOTALL);
+
+	@Override
+	public void changedAttributeHook(PerunSessionImpl session, Member member, Attribute attribute) throws InternalErrorException, WrongReferenceAttributeValueException {
+		if (attribute.getValue() == null) {
+			return;
+		}
+
+		Map<String, String> valueOfAttribute = (Map<String, String>) attribute.getValue();
+
+		if (valueOfAttribute.isEmpty()) {
+			return;
+		}
+
+		if (!valueOfAttribute.containsKey("userId") || !valueOfAttribute.containsKey("timestamp")) {
+			int id = session.getPerunPrincipal().getUser().getId();
+			valueOfAttribute.put("userId", String.valueOf(id));
+			Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+			valueOfAttribute.put("timestamp", timestamp.toString());
+			try {
+				session.getPerunBl().getAttributesManagerBl().setAttribute(session, member, attribute);
+			} catch (WrongAttributeValueException | WrongAttributeAssignmentException e) {
+				throw new ConsistencyErrorException(e);
+			}
+		}
+	}
+
+	@Override
+	public List<String> resolveVirtualAttributeValueChange(PerunSessionImpl session, String message) throws WrongReferenceAttributeValueException, AttributeNotExistsException, WrongAttributeAssignmentException {
+
+		Matcher statusChangeMatcher = statusChange.matcher(message);
+
+		if (statusChangeMatcher.find() && (statusChangeMatcher.group(2).equals("expired") || statusChangeMatcher.group(2).equals("disabled") || statusChangeMatcher.group(2).equals("validated"))) {
+			try {
+				int memberId = Integer.valueOf(statusChangeMatcher.group(1));
+				Member member = session.getPerunBl().getMembersManager().getMemberById(session, memberId);
+				Attribute attribute = session.getPerunBl().getAttributesManagerBl().getAttribute(session, member, AttributesManager.NS_MEMBER_ATTR_DEF + ":reasonToSuspend");
+				attribute.setValue(new HashMap<String, String>());
+				session.getPerunBl().getAttributesManagerBl().setAttribute(session, member, attribute);
+			} catch (Exception e) {
```

Comment:
Dont catch all exceptions like that. First, catch AttributeNotExistsException if this exception is thrown, it means that member's status has been changed but he did not have 'reasonToSuspend' attribute set and that is not required. So, just catch that exception and put a comment inside of this catch block.

Then catch all other excpetions by their specific names and log them the way you already do.